docs/DeveloperGuide.md

```diff
@@ -1,34 +1,178 @@
 # Developer Guide
 
-## Design & implementation
+![Supported Java versions](https://img.shields.io/badge/Java-11-blue.svg)
+![Supported OS](https://img.shields.io/badge/Supported%20OS-Windows|MacOS|Linux-yellow.svg)
+![Java CI](https://github.com/AY2021S1-CS2113T-T09-4/tp/workflows/Java%20CI/badge.svg)
+
+## Table of Contents
+
+- [Introduction](#introduction)
+    * [To be added](#to-be-added)
+        + [To be added](#to-be-added)
+
+## 1. Introduction
+
+### 1.1 Software Overview
+
+easyLog is a Command Line Interface (CLI) application for warehouse employees to manage items and orders in the
+warehouse. easyLog allows users to add new items and orders, view existing items and orders, remove specific items and
+orders and clear all existing items and orders in the warehouse at ease. If the user types fast, easyLog can get
+logistic management tasks done faster than traditional Graphical User Interface (GUI) applications.
+
+easyLog, comprises a Parser component, Ui component, Storage component, Item component, Order component and Command
+component. Each component consists of various classes that work in tandem, to ensure it meets the purpose of our
+program.
+
+The purpose of this developer guide is to allow any interested contributors, who wish to develop this application
+further or just curious about the workings of this application. This would allow potential contributors to be able to
+dive right in to improving the code, performance, features or even adding new features much more easily due to the
+understanding of the structure of the codebase and implementation of existing features.
+
+### 1.2 Prerequisites
+
+- Java 11 (can be downloaded from [here](https://www.oracle.com/sg/java/technologies/javase-jdk11-downloads.html))
+- IntelliJ Idea (can be downloaded from [here](https://www.jetbrains.com/idea/download/#section=mac))
```

Comment:
I like that you included links to the prerequisite software. Really helps someone new who is not tech savvy to figure out how to set your app up.